src/wamr/mpi.cpp

```diff
@@ -61,37 +61,37 @@ class WamrMpiContextWrapper
     }
 
     // MPI passes an MPI_Request* as part of the asynchronous API calls.
-    // MPI_Request is in itself a faabric_request_t* so, to write a value to
-    // it, we'd have to allocate memory for the faabric_reques_t. To aovid
-    // doing that, we write the actual request id to a faabric_reques_t*.
-    void writeFaasmRequestId(int32_t* requestPtrPtr, int32_t requestId)
+    // MPI_Request is in itself a faabric_request_t* so requestPtrPtr is a
+    // faabric_request_t**, which is a double wasm offset. Allocating memory
+    // for the second pointer from outside WASM is tricky and error-prone,
+    // so we use overwrite the pointer value with the actual pointed-to value
+    // (the pointer just points to one int, so we can do that).
+    void writeFaasmRequestId(int32_t* requestPtrPtr, int32_t requestId) const
     {
         module->validateNativePointer(requestPtrPtr, sizeof(MPI_Request));
         MPI_Request* requestPtr = reinterpret_cast<MPI_Request*>(requestPtrPtr);
-        faabric::util::unalignedWrite<faabric_request_t*>(
```

Comment:
This `unalignedWrite` was actually smashing some values in the WASM stack, what was making LAMMPS hang mid execution, and made me have a very entretaining week :rofl: